code/game/machinery/engi_adv.dm

```diff
@@ -0,0 +1,240 @@
+// Construction "nuke"
+
+/obj/machinery/construction_nuke
+	name = "nuclear fission construction device"
+	desc = "The next level of interior redecoration."
+	icon = 'icons/obj/machines/nuke.dmi'
+	icon_state = "nuclearbomb0"
+	density = TRUE
+	use_power = 0
+
+	var/timer_set = 90
+	var/ui_style = "nanotrasen"
+	var/range = 160
+
+	var/timing = FALSE
+	var/detonation_timer = null
+	var/cooldown = 0
+	var/safety = TRUE
+
+	var/bomb_set = FALSE
+	var/exploding = FALSE
+	var/quiet = FALSE
+	var/payload = "plasteel"
+	var/payload_wall = /turf/closed/wall/r_wall
+	var/payload_floor = /turf/open/floor/engine
+	var/static/list/possible_payloads = list("wood","sand","ice","mining","silver","gold","bananium","abductor","desolation", "plasma","uranium","bluespace","diamond","plasteel","safety","titanium","plastitanium", )
```

Comment:
Won't the comma at the end here cause problems?  Yet it still compiled.  *shrug*